### ISSUE TITLE
Main window blanks out when changing from Prometheus datasource type

### ISSUE THREAD

**Issue by adeverteuil** [grafana/grafana#18842]
**What happened**:

We found a way to cause the main window to blank out.

**How to reproduce it (as minimally and precisely as possible)**:

1. Select a **Prometheus** datasource.
2. Write any trivial query: `avg(rate(node_cpu_seconds_total{mode!="idle"}[5m]))`.
3. Select **any other** datasource type.
4. *main div blanks out*

The problem doesn't happen if the query is empty or if you select another Prometheus datasource.

**Anything else we need to know?**:

Screenshot with DOM explorer open and the empty `div` highlighted:

![screenshot 2019-09-02 1654](https://user-images.githubusercontent.com/3270755/64133521-a2225380-cda4-11e9-99aa-38f2039ca7e6.png)

2 Javascript errors in browser console:

```
TypeError: "e is undefined"
    q dom.ts:20
    $ dom.ts:39
    componentDidUpdate QueryField.tsx:532
    React 3
    unstable_runWithPriority scheduler.production.min.js:18
    React 5
react-dom.production.min.js:198:186
```
```
TypeError: e is undefined[Learn More] dom.ts:20:6
```

**Environment**:
- Grafana version: 6.3.4 and 6.3.5.
- Data source type & version: Prometheus
- OS Grafana is installed on: Official docker image
- User OS & Browser:
- Grafana plugins:
- Others:

**Comment by davkal** [comment]
Can repro in 6.3.3, somehow componentDidUpdate is called with an empty query on the query field before it gets removed. This was fixed in #18599, so needs to be cherry-picked if we want to release this.

**Comment by marefr** [comment]
Fixed by #18599
